Remotion/Validation/Core/Utilities/TopologySortExtensions.cs

```diff
@@ -206,15 +216,16 @@ public static IEnumerable<IEnumerable<T>> TopologySortDesc<T> (this IEnumerable<
     public static IEnumerable<IEnumerable<T>> TopologySort<T> (
         this IEnumerable<T> source,
         Func<T, IEnumerable<T>> getDependencies,
-        Func<IEnumerable<T>, IEnumerable<T>> subSort,
+        Func<IEnumerable<T>, IEnumerable<T>>? subSort,
         TopologySortMissingDependencyBehavior missingDependencies
         )
+        where T : notnull
     {
       ArgumentUtility.CheckNotNull ("source", source);
       ArgumentUtility.CheckNotNull ("getDependencies", getDependencies);
 
       var unsorted = source.Select (content => new Node<T> (content, getDependencies, true)).ToList();
-      Dictionary<object, Node<T>> nodes;
+      Dictionary<object, Node<T>>? nodes;
```

Comment:
Same as with the attributes above, I added a rule for uninitialized local variables because I thought they would default to null instead of yielding a compiler error on use.